Sources/SwiftFormat/LintPipeline.swift

```diff
@@ -20,38 +20,35 @@ extension LintPipeline {
   /// Calls the `visit` method of a rule for the given node if that rule is enabled for the node.
   ///
   /// - Parameters:
-  ///   - visitor: A reference to the `visit` method on the *type* of a `SyntaxLintRule` subclass.
-  ///     The type of the rule in question is inferred from the signature of the method reference.
-  ///   - context: The formatter context that contains information about which rules are enabled or
-  ///     disabled.
+  ///   - visitor: A reference to the `visit` method on an instance of a `SyntaxLintRule` subclass.
   ///   - node: The syntax node on which the rule will be applied. This lets us check whether the
   ///     rule is enabled for the particular source range where the node occurs.
-  func visitIfEnabled<Rule: SyntaxLintRule, Node: SyntaxProtocol>(
-    _ visitor: (Rule) -> (Node) -> SyntaxVisitorContinueKind, in context: Context, for node: Node
+  ///   - name: The name of the rule from the `SyntaxLintRule` subclass. Used to determine if the
+  ///     rule should be applied to the current node.
+  func visitIfEnabled<Node: SyntaxProtocol>(
```

Comment:
Is it possible to keep the signature of these methods the same to avoid the repetition and churn to the generator?

Since we have `Rule` as a generic argument that's inferred from the `visit` method that gets passed in, it seems like we should be able to replace the `Rule(context:)` initialization in this function with a call to the `rule` helper function to get the cached one, but leave everything else more or less the same.

Reply:
Good point. I mostly reverted the signatures of these methods, but I still deleted the `context` param because it's not used and it's already available as a property of `LintPipeline` anyway.

Reply:
Even better!